Content.Server/Spawners/EntitySystems/ConditionalSpawnerSystem.cs

```diff
@@ -2,24 +2,110 @@
 using Content.Server.Spawners.Components;
 using JetBrains.Annotations;
 using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
 
 namespace Content.Server.Spawners.EntitySystems
 {
     [UsedImplicitly]
     public class ConditionalSpawnerSystem : EntitySystem
     {
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly GameTicker _ticker = default!;
+
         public override void Initialize()
         {
             base.Initialize();
 
             SubscribeLocalEvent<GameRuleAddedEvent>(OnRuleAdded);
+            SubscribeLocalEvent<ConditionalSpawnerComponent, MapInitEvent>(OnCondSpawnMapInit);
+            SubscribeLocalEvent<RandomSpawnerComponent, MapInitEvent>(OnRandSpawnMapInit);
+        }
+
+        private void OnCondSpawnMapInit(EntityUid uid, ConditionalSpawnerComponent component, MapInitEvent args)
+        {
+            TrySpawn(component);
+        }
+
+        private void OnRandSpawnMapInit(EntityUid uid, RandomSpawnerComponent component, MapInitEvent args)
+        {
+            Spawn(component);
+            EntityManager.QueueDeleteEntity(uid);
         }
 
         private void OnRuleAdded(GameRuleAddedEvent args)
         {
-            foreach (var spawner in EntityManager.EntityQuery<ConditionalSpawnerComponent>())
```

Comment:
I think this is supposed to exclude paused entities. so that game rules can be configured before map init?